modin/numpy/arr.py

```diff
@@ -1347,6 +1347,60 @@ def __rmul__(
     ):
         return self.multiply(x2, out, where, casting, order, dtype, subok)
 
+    def dot(self, other, out=None):
```

Comment:
We need to alias `__matmul__` to `dot`.

Reply:
The behavior of `matmul` and `dot` are not quite the same: https://numpy.org/doc/stable/reference/generated/numpy.matmul.html

That said, it looks like the "stacks of matrices" case only applies for higher dimensions, and the scalar multiplication case is easy to special case out. I'll add this extra logic.